src/main/angular/app/services/data.service.ts

```diff
@@ -11,4 +11,6 @@ export abstract class DataService {
   abstract async getProjectGraphData(id: string): Promise<ProjectGraphData>;
   /** Returns whether there is at least one pending web request. */
   abstract hasPendingRequest(): boolean;
```

Comment:
As a general principle, IMO, more granular is better. 
E.g. Imagine a future where Admin components were moved to the main view page - it would show loading when project details were being fetched even though a different RPC was in progress.

Consider refactoring this method to hasPendingRequest(RequestType) so that different areas of the UI can differentiate what is in flight, but it is not required. I would prioritize VM recommendations work over this at this time.

Reply:
Makes sense. I'll mark this down as a low-priority TODO and will implement it if I have the time.